app/controllers/kudos_controller.rb

```diff
@@ -55,4 +56,8 @@ def set_kudo
   def kudo_params
     params.require(:kudo).permit(:title, :content, :receiver_id, :giver_id)
   end
+
+  def kudo_giver_check
+    return redirect_to root_path, notice: 'You are not authorized to edit this kudo.' if @kudo.giver != current_employee
```

Comment:
are you sure you need this return? I'd check what would happen without it :)

Reply:
That's right, after removing return everything is OK.